R/simulate.r

```diff
@@ -1,52 +1,37 @@
-#' Simulate transmission chains using a branching process
-#'
-#' @description \code{chain_sim()} is a stochastic simulator for generating
-#' transmission chain data with key inputs such as the offspring distribution
-#' and serial interval distribution.
-#' @param n Number of simulations to run.
-#' @param offspring Offspring distribution: a character string corresponding to
-#'   the R distribution function (e.g., "pois" for Poisson, where
-#'   \code{\link{rpois}} is the R function to generate Poisson random numbers)
-#' @param stat String; Statistic to calculate. Can be one of:
+#' Simulate a tree of infections with a serial and offspring distributions
+#'
+#' @param nchains Number of chains to simulate.
+#' @param offspring_sampler Offspring distribution: a character string
+#' corresponding to the R distribution function (e.g., "pois" for Poisson,
+#' where \code{\link{rpois}} is the R function to generate Poisson random
+#' numbers)
+#' @param chain_statistic String; Statistic to calculate. Can be one of:
 #' \itemize{
 #'   \item "size": the total number of offspring.
 #'   \item "length": the total number of ancestors.
 #' }
-#' @param infinite A size or length above which the simulation results
-#' should be set to `Inf`. Defaults to `Inf`, resulting in no results
-#' ever set to `Inf`
-#' @param tree Logical. Should the transmission tree be returned? Defaults
-#' to `FALSE`.
-#' @param serial The serial interval generator function; the name of a
+#' @param chain_stat_max A cut off for the chain statistic (size/length) being
+#' computed. Results above the specified value, are set to this value.
+#' Defaults to `Inf`.
+#' @param serials_sampler The serial interval generator function; the name of a
```

Comment:
perhaps rename it something that contains "interval", e.g. `si_sampler` (though see also my other comment on supporting character strings vs. anonymous functions here - I think there might be an argument for just supporting character strings in which case this would be `si_dist`)

Reply:
I think the result of the discussion (including https://github.com/epiverse-trace/epichains/issues/25#issuecomment-1651715438) is that we want to do this via character strings everywhere instead of functions (though happy to be told otherwise if you read this differently). Perhaps convert my previous comment to an issue for future processing.